compiler_gym/envs/compiler_env.py

```diff
@@ -193,8 +193,8 @@ def __init__(
         self,
         service: Union[str, Path],
         benchmark: Optional[Union[str, Benchmark]] = None,
-        observation_space: Optional[str] = None,
-        reward_space: Optional[str] = None,
+        observation_space: Optional[Union[str, ObservationSpaceSpec]] = None,
+        reward_space: Optional[Union[str, RewardSpaceSpec]] = None,
```

Comment:
Good catch, I've updated the docstrings. A user can make their own observation spaces using [add_derived_space()](https://facebookresearch.github.io/CompilerGym/compiler_gym/views.html#compiler_gym.views.ObservationView.add_derived_space).

Action spaces don't have a spec because at the moment they are simply `gym.spaces.Discrete` categorical spaces. Once we add support for other space types that will be updated